### DISCUSSION TITLE
Estimated Battery Life Calculator

### DISCUSSION THREAD

**Discussion by Homer-Rivera** [jasonacox/Powerwall-Dashboard#39] [Ideas]
It would be cool to add a battery life estimation based on the current powerwall load when you are running on your backup batteries to give you an idea of how long they will last.

BATTERY LIFE FORMULA = Battery Capacity(mAh) / Load Current(mAh)

**Comment by youzer-name** [comment]
I've been playing around with this a bit and it wouldn't be too hard to add a panel with the formula you listed, although it does involve using transform operations in Grafana to merge the battery data and the load data.

However I'm not sure there is a simple way to get a meaningful answer, unless you have a Powerwall with no solar. Also, I don't think you really want to use current load. You probably want to average that out over at least a short time period.

For example you could simply take (current battery capacity in kWh) / (average load for the last hour) and that would give you the number of hours the battery would last without any solar input. For me that currently is (12.1 kWh total in two Powerwalls @ 38% charge) / .961 kWh consumed in the last hour) = 12.6 hours of battery. 

That is totally unrealistic as it is going to be a hot day and my AC is going to increase the house load a lot over the next 12 hours, but that is the math.

However if I include the solar production from the last 24 hours:

(12.2 kWh currently in the batteries + 33.7 kWh of solar generated in the last 24 hours) / 40.7 kWh of home use in the last 24 hours  = 1.13 days of power. Anything above 1.0 is good here, as it means I'd make it into tomorrow if today is similar to yesterday in use and production.

So if the sun went out, I can go about 13 hours based on my current battery and my last hour's consumption. If I account for solar production, I can go > 24 hours based on current battery and solar production. I'm not sure either of these stats is really useful because there are so many variables that can make them turn out to be incorrect.

**Reply by Homer-Rivera** [reply]
My thought was the result would show infinity if the system is running with the grid.  This would be a metric that would start calculating if the grid went offline and the solar output was 0.  Now that you are just running on your backup battery you can see based on your current draw how long you would last.  Then I could play with turning on and off devices to see how it would affect me lasting through the night till the next day.
